test/long_stream_test.go

```diff
@@ -0,0 +1,220 @@
+//go:build !fast
+// +build !fast
+
+package test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/ydb-platform/ydb-go-sdk/v3/table/options"
+	"io"
+	"os"
+	"path"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	ydb "github.com/ydb-platform/ydb-go-sdk/v3"
+	"github.com/ydb-platform/ydb-go-sdk/v3/table"
+	"github.com/ydb-platform/ydb-go-sdk/v3/table/types"
+)
+
+func TestLongStream(t *testing.T) {
+	var (
+		discoveryInterval = 10 * time.Second
+		db                ydb.Connection
+		err               error
+		upsertRowsCount   = 10000
+		batchSize         = 1000
+	)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+	defer cancel()
+
+	t.Run("make connection", func(t *testing.T) {
+		db, err = ydb.Open(
+			ctx,
+			os.Getenv("YDB_CONNECTION_STRING"),
+			ydb.WithAccessTokenCredentials(
+				os.Getenv("YDB_ACCESS_TOKEN_CREDENTIALS"),
+			),
+			ydb.WithDiscoveryInterval(0), // disable re-discovery on upsert time
+		)
+		if err != nil {
+			t.Fatal(err)
+		}
+	})
+
+	defer func(db ydb.Connection) {
+		// cleanup
+		_ = db.Close(ctx)
+	}(db)
+
+	t.Run("creating stream table", func(t *testing.T) {
+		if err = db.Table().Do(
+			ctx,
+			func(ctx context.Context, s table.Session) (err error) {
+				return s.ExecuteSchemeQuery(
+					ctx,
+					`CREATE TABLE long_stream_query (val Int32, PRIMARY KEY (val))`,
+				)
+			},
+		); err != nil {
+			t.Fatalf("create table failed: %v\n", err)
+		}
+	})
+
+	t.Run("check batch size", func(t *testing.T) {
+		if upsertRowsCount%batchSize != 0 {
+			t.Fatalf("wrong batch size: (%d mod %d = %d) != 0", upsertRowsCount, batchSize, upsertRowsCount%batchSize)
+		}
+	})
+
+	t.Run("upserting rows", func(t *testing.T) {
+		var upserted uint32 = 0
+		for i := 0; i < (upsertRowsCount / batchSize); i++ {
+			t.Run("", func(t *testing.T) {
+				values := make([]types.Value, 0, upsertRowsCount)
+				for j := 0; j < upsertRowsCount; j++ {
```

Comment:
It should be `batchSize`, not `upsertRowsCount`.

Reply:
ack